python/lsst/ci/models.py

```diff
@@ -0,0 +1,138 @@
+from dataclasses import dataclass
+from typing import List, Set, Optional, OrderedDict as ODict
+from collections import OrderedDict
+
+from lsst.ci import tsort
+
+
+class ProductIndex(OrderedDict):
+    """
+    Container class for a product index with index-related operations.
```

Comment:
Please move the documentation for `__init__` to this doc string as per our standard: https://developer.lsst.io/python/numpydoc.html#py-docstring-class-structure

Please look for this same issue elsewhere

Also (picky but) please remove the \n after the opening """. If necessary you can wrap the brief description.